### PR TITLE
Can recommend taxi reservation deep links now

### LINKED ISSUE #36: [Feature] Add location recommendation
Add taxi reservation deep links into recommendation logic.

### PR DESCRIPTION
**Pull request by RazorBach**
Add taxi reservation recommendation logic into app: when user type a sentence. Use text classifier to identify if the text is an address. And use geocoder to get the coordinates and update the view.

fix #36

### CODE REVIEW

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/constants/Constant.java [hunk 1]
```diff
@@ -10,54 +9,84 @@
...
+              "actions.intent.CREATE_TAXI_RESERVATION",
+              "actions.intent.GET_TAXI_RESERVATION"));
 
   @StringRes
-  public static final int[] TAB_TITLES = new int[]{R.string.tab_text_statistics, R.string.tab_text_all, R.string.tab_text_1, R.string.tab_text_2, R.string.tab_text_3, R.string.tab_text_4, R.string.tab_text_5};
+  public static final int[] TAB_TITLES =
+      new int[] {
+        R.string.tab_text_statistics,
+        R.string.tab_text_all,
+        R.string.tab_text_1,
```
Comment (hilaryhuo) [inline AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/constants/Constant.java]: would you want to rename this to something like tab_text_common maybe? and same for the following

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java [hunk 2]
```diff
@@ -51,11 +57,13 @@ public static int getResIdByPackageName(String pkgName, Class<?> className) {
   }
 
   /**
-   * @param context
+   * get app's Name by packageName 1. Use packageManager 2. use built-in * resources(currently
+   * support 56 apps). 3. if 1 and 2 cannot find, return "*unknown" (* is to * make user it will be
```
Comment (hilaryhuo) [inline AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java]: nit: "* is to make " ? extra *

**Review by hilaryhuo — COMMENTED** [review]
left some minor comments, everything else looks good

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java [hunk 3]
```diff
@@ -108,11 +116,15 @@ public static void jumpToWebPage(Context context, final String url) {
 
   // first check if package exists since user may haven't installed the app
   // then check if the url can jump to activity
-  public static void jumpToApp(Context context, final String url, final String packageName) {
+  public static void jumpToApp(Context context, String url, final String packageName) {
+    if (packageName.equals("com.dunkinbrands.otgo")) {
```
Comment (jianjinjian) [inline AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java:121]: some comment on the special handling here

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/activity/ParameterActivity.java [hunk 4]
```diff
@@ -267,55 +265,33 @@ void replaceAddressParameter(Intent data) {
...
-          getString(
-              R.string.url_parameter,
-              entry.getKey(),
-              data.getStringExtra(Constant.DROP_OFF_LONGITUDE)));
-    }
-  }
-
-
   /**
+   * replace each parameter with input from user to construct the url e.g.: *
```
Comment (jianjinjian) [inline AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/activity/ParameterActivity.java]: nit

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/constants/Constant.java (hunk 1 above)
Comment (RazorBach) [reply AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/constants/Constant.java]: Sure, fixed.

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java (hunk 2 above)
Comment (RazorBach) [reply AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java]: Oh, I didn't check this comment after formatting. Fixed now

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java (hunk 3 above)
Comment (RazorBach) [reply AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/AppUtils.java:121]: Sure, fixed

File: AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/activity/ParameterActivity.java (hunk 4 above)
Comment (RazorBach) [reply AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/activity/ParameterActivity.java]: fixed

**Comment by RazorBach** [comment]
All comments fixed. I also check into some gradlew and wrapper so anyone can work with it without needing to install Gradle beforehand.

**Review by jianjinjian — APPROVED** [review]

**Pull request merged**